reference-implementation/__tests__/composition.js

```diff
@@ -0,0 +1,367 @@
+      {
+        imports: { 'https://a/': 'https://b/' }
+      },
+      {
+        imports: { 'https://c/': 'https://d/' }
+      },
+      {
+        imports: { 'https://e/': 'https://f/' }
+      }
+    ])).toStrictEqual({
+      imports: {
+        'https://a/': ['https://b/'],
+        'https://c/': ['https://d/'],
+        'https://e/': ['https://f/']
+      },
+      scopes: {}
+    });
+  });
+
+  it('should compose maps that interact via cascading', () => {
+    expect(composeMaps([
+      {
+        imports: { 'https://c/': 'https://d/' }
+      },
+      {
+        imports: { 'https://b/': 'https://c/' }
+      },
+      {
+        imports: { 'https://a/': 'https://b/' }
+      }
+    ])).toStrictEqual({
+      imports: {
+        'https://a/': ['https://d/'],
+        'https://b/': ['https://d/'],
+        'https://c/': ['https://d/']
+      },
+      scopes: {}
+    });
+  });
+
+  it('should compose maps with fallbacks that interact via cascading', () => {
+    expect(composeMaps([
+      {
+        imports: { 'https://e/': ['https://g/', 'https://h/'] }
+      },
+      {
+        imports: {
+          'https://c/': ['https://f/'],
+          'https://b/': ['https://d/', 'https://e/']
+        }
+      },
+      {
+        imports: { 'https://a/': ['https://b/', 'https://c/'] }
+      }
+    ])).toStrictEqual({
+      imports: {
+        'https://a/': ['https://d/', 'https://g/', 'https://h/', 'https://f/'],
+        'https://b/': ['https://d/', 'https://g/', 'https://h/'],
+        'https://c/': ['https://f/'],
+        'https://e/': ['https://g/', 'https://h/']
+      },
+      scopes: {}
+    });
+  });
+
+  it('should compose maps that are using the virtualisation patterns we expect to see in the wild', () => {
+    expect(composeMaps([
+      {
+        imports: {
+          'std:built-in': 'https://built-in-enhancement-1/'
+        },
+        scopes: {
+          'https://built-in-enhancement-1/': {
+            'std:built-in': 'std:built-in'
+          }
+        }
+      },
+      {
+        imports: {
+          'std:built-in': 'https://built-in-enhancement-2/'
+        },
+        scopes: {
+          'https://built-in-enhancement-2/': {
+            'std:built-in': 'std:built-in'
+          }
+        }
+      },
+      {
+        imports: {
+          'std:built-in': 'https://built-in-enhancement-3/'
+        },
+        scopes: {
+          'https://built-in-enhancement-3/': {
+            'std:built-in': 'std:built-in'
+          }
+        }
+      }
+    ])).toStrictEqual({
+      imports: { 'std:built-in': ['https://built-in-enhancement-3/'] },
+      scopes: {
+        'https://built-in-enhancement-1/': { 'std:built-in': ['std:built-in'] },
+        'https://built-in-enhancement-2/': { 'std:built-in': ['https://built-in-enhancement-1/'] },
+        'https://built-in-enhancement-3/': { 'std:built-in': ['https://built-in-enhancement-2/'] }
+      }
+    });
+  });
+
+  it('should compose "nested" scopes', () => {
```

Comment:
This seems to make a decision on https://github.com/WICG/import-maps/issues/153. Which I'm fine with; that issue is unresolved just because I've been waiting for someone to chime in, and you've done it. But can we add a simple test, like the one from (the scopes part of) https://github.com/WICG/import-maps/issues/153#issuecomment-513843388 , before getting in to cascading?